src/runtime/pkg/kata-monitor/shim_client.go

```diff
@@ -36,6 +37,13 @@ func getSandboxIDFromReq(r *http.Request) (string, error) {
 	return "", fmt.Errorf("sandbox not found in %+v", r.URL.Query())
 }
 
+func checkShimSocketExists(sandboxID string) bool {
+	socketPath := shim.SocketAddress(sandboxID)
```

Comment:
Does this work for CRI-O too?

Reply:
Hey @liubin , thanks for checking!
This should work for *any* container engine! We check if there is the shimv2 socket for the POD (that we later use to retrieve the metrics), if so we are pretty sure we are dealing with a kata pod. Wondering... should we drop the check on annotations for CRI-O and containerd and keep just this check?

I took some time to reply because your question hinted me something should be wrong... I prepared a test environment to check this case and debugged: found that I forgot to trim the "unix://" prefix of the socket before checking if the socket file existed (obviously). Pushed a fixed version for that.
I am using the shim.SocketAddress function here because I don't want to hardcode the path here: let's use the path from the containerd-shimv2 code: so if it will change there for any reason, we should be safe here.